src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -103,6 +106,88 @@ internal sealed partial class LocalRewriter
 
                 return _factory.Sequence(temps.ToImmutableAndFree(), effects.ToImmutableAndFree(), returnValue);
             }
+
+            void reverseAssignmentsToTargetsIfApplicable()
+            {
+                // The list of deconstruction targets is usually small. Prefer a temporary array which is stack-bound for small numbers of elements.
+                var visitedSymbols = TemporaryArray<Symbol>.Empty;
+
+                if (right is BoundConvertedTupleLiteral or BoundConversion { Operand.Kind: BoundKind.TupleLiteral or BoundKind.ConvertedTupleLiteral }
```

Comment:
>BoundKind.TupleLiteral or BoundKind.ConvertedTupleLiteral [](http://example.com/codeflow?start=91&length=57)

Do tests cover both options?

Reply:
Looked at this a little closer. I think testing for BoundKind.TupleLiteral is unnecessary. Removing that subpattern and adding an assertion that this is never the kind of the RHS.